guides/keras_cv/random_walks_with_stable_diffusion.py

```diff
@@ -0,0 +1,247 @@
+the gif.
+"""
+
+def export_as_gif(filename, images, frames_per_second=20, rubber_band=False):
+    if rubber_band:
+        images += images[2:-1][::-1]
+    images[0].save(filename, save_all=True, append_images=images[1:],
+                   duration=1000 // frames_per_second, loop=0)
+
+export_as_gif("doggo-and-fruit-5.gif", [Image.fromarray(img) for img in images],
+              frames_per_second=2, rubber_band=True)
+
+"""
+![Dog to Fruit 5](https://imgur.com/a/617Qjxb)
+
+The results may seem surprising. Generally, interpolating between prompts
+produces coherent looking images, and often demonstrate a progressive concept
+shift between the contents of the two prompts.
+
+To best visualize this, we should do a much more fine-grained interpolation,
+using hundreds of steps. In order to keep batch size small (so that we don't
+OOM our accelerators), this requires manually batching our interpolated
+encodings.
+"""
+
+interpolation_steps = 160
+batch_size = 16
+
+encoding_step = (encoding_2 - encoding_1) / interpolation_steps
+encoding_step = tf.squeeze(encoding_step)
+
+images = []
+for step_index in range(interpolation_steps // batch_size):
+    step_offsets = tf.linspace(batch_size*step_index,
+                               batch_size*(step_index+1) - 1, batch_size)
+    step_offsets = tf.tensordot(tf.cast(step_offsets, encoding_step.dtype),
+                                encoding_step, 0)
+    encodings = encoding_1 + tf.cast(step_offsets, encoding_1.dtype)
+    images += [Image.fromarray(img) for img in model.generate_image(
+        encodings,
+        batch_size=batch_size,
+        num_steps=25,
+        diffusion_noise=noise,
+    )]
+
+export_as_gif("doggo-and-fruit-160.gif", images, rubber_band=True)
+
+"""
+![Dog to Fruit 160](https://imgur.com/a/iNfRnH4)
+
+The resulting gif shows a much clearer and more coherent shift between the two
+prompts. Try out some prompts of your own and experiment!
+
+## A Random Walk Around a Text Prompt
+
+Our next experiment will be to go for a random walk around the latent manifold
+starting from a point produced by a particular prompt.
+"""
+
+walk_steps = 160
+batch_size = 16
+step_size = .05
+
+encoding = model.encode_text("A rubber duck swimming in a bowl of cereal")
+
+images = []
+for step_index in range(walk_steps // batch_size):
+    batch_encodings = []
+    for individual_image_step in range(batch_size):
+        batch_encodings.append(encoding)
+        # Note that (77, 768) is the shape of the text encoding.
+        random_deltas = tf.random.normal((77, 768), stddev=step_size, seed=seed)
+        encoding += random_deltas
+    batch_encodings = tf.stack(batch_encodings)
+    images += [Image.fromarray(img) for img in model.generate_image(
+        batch_encodings,
+        batch_size=batch_size,
+        num_steps=25,
+        diffusion_noise=noise,
+    )]
+
+export_as_gif("ducky.gif", images, rubber_band=True)
+
+"""
+![Rubber Ducky Goes on a Walk](https://imgur.com/a/hZAlHrz)
```

Comment:
Let’s slow these gifs down a bit - the latent space is super smooth and it would be cool to see that

Reply:
I'm re-running now to slow these down (and use a new seed, because I realized that my original gifs were created with a random seed and I want them to be reproducible with this exact code)